src/tests/multihost/ad/test_adparameters_ported.py

```diff
@@ -0,0 +1,1749 @@
+    )
+
+    # Add member
+    session_multihost.client[0].run_command(
+        f'adcli add-member -D {domainname} {ad_group} {ad_user}',
+        stdin_text=password, raiseonerr=False
+    )
+
+    def remove_plain_ad_user_group():
+        """ Remove windows AD user and group """
+        ad = ADOperations(session_multihost.ad[0])
+        ad.delete_ad_user_group(ad_group)
+        ad.delete_ad_user_group(ad_user)
+
+    request.addfinalizer(remove_plain_ad_user_group)
+    return ad_user, ad_group
+
+
+@pytest.fixture(scope="class")
+def change_client_hostname(session_multihost, request):
+    """ Change client hostname to a truncated version in the AD domain"""
+    cmd = session_multihost.client[0].run_command('hostname', raiseonerr=False)
+    old_hostname = cmd.stdout_text.rstrip()
+    ad_domain = session_multihost.ad[0].domainname
+    try:
+        new_hostname = session_multihost.client[0].external_hostname.\
+            split('.')[0]
+    except (KeyError, AttributeError):
+        new_hostname = old_hostname.split('.')[0]
+    if new_hostname.startswith('ci-'):
+        new_hostname = new_hostname[3:]
+    new_hostname = new_hostname[:15] + "." + ad_domain
+    session_multihost.client[0].run_command(
+        f'hostname {new_hostname}', raiseonerr=False
+    )
+
+    def restore():
+        """ Restore hostname """
+        session_multihost.client[0].run_command(
+            f'hostname {old_hostname}',
+            raiseonerr=False
+        )
+    request.addfinalizer(restore)
+
+
+@pytest.mark.adparameters
+@pytest.mark.usefixtures("change_client_hostname")
+class TestADParamsPorted():
+    """ BZ Automated Test Cases for AD Parameters ported from bash"""
+
+    @pytest.fixture(autouse=True, scope="class")
+    def _setup(self, session_multihost):
+        self.ad_realm = session_multihost.ad[0].domainname.upper()
+        self.ad_domain_short = self.ad_realm.rsplit('.', 1)[0]
+
+    @staticmethod
+    @pytest.mark.tier1
+    def test_0001_ad_parameters_domain(multihost, adjoin, create_aduser_group):
+        """
+        :title: IDM-SSSD-TC: ad_provider: ad_parameters: Set ad domain to
+          AD DOMAIN1
+        :id: 08a256e6-a56e-4726-adba-b9093dce8ede
+        :setup:
+         1. Configure short domain name, clear cache and restart sssd.
+         2. Create AD user and group.
+        :steps:
+          1. Run getent passwd for the user and group
+          2. Run getent group for the group
+          3. Run check that su can switch to the ad user in short domain
+          4. Check the sssd domain log
+        :expectedresults:
+          1. User is found
+          2. Group is found
+          3. Su works as expected
+          4. Log contains the expected lines
+             Option ad_domain has value ...
+             Option krb5_realm set to ...
+        :customerscenario: False
+        """
+        adjoin(membersw='adcli')
+        # Create AD user and group
```

Comment:
I feel these comments in the test cases are redundant, the description is great, so it's easy to follow the code knowing what it's supposed to do. I think we should remove it, but I'll leave this up to you.

Reply:
I will definitely need to go through the comments and descriptions in detail in the future to make sure that the test steps are properly describing what is going on to avoid copy&paste isues. I can remove redundant code comments afterwards.